packages/server/app/Controllers/Ws/MpeRoomsWsController.ts

```diff
@@ -407,4 +488,68 @@ export default class MpeRoomsWsController {
             throw new Error('onExportToMtv error');
         }
     }
+
+    public static async onCreatorInviteUser({
+        invitedUserID,
+        invitingUserID,
+        roomID,
+    }: OnCreatorInviteUserArgs): Promise<void> {
+        const room = await MpeRoom.findOrFail(roomID);
+        const invitedUser = await User.findOrFail(invitedUserID);
+
+        const userIsNotRoomCreator = invitingUserID !== room.creatorID;
+
+        if (userIsNotRoomCreator) {
+            throw new Error(
+                `Emitter user does not appear to be the mpe room creator`,
+            );
+        }
+
+        const creatorIsInvitingHimself = invitedUser.uuid === invitingUserID;
+        if (creatorIsInvitingHimself) {
+            throw new Error('Creator cannot invite himself in his mpe room');
+        }
+
+        await invitedUser.load('mpeRooms', (mpeRoomQuery) => {
+            return mpeRoomQuery.where('uuid', roomID);
+        });
+        const invitedUserIsAlreadyInTheRoom =
+            invitedUser.mpeRooms !== null && invitedUser.mpeRooms.length === 1;
+        if (invitedUserIsAlreadyInTheRoom) {
+            throw new Error('Invited user is already in the mpe room');
+        }
+
+        await room.load('creator');
+        if (room.creator === null) {
+            throw new Error(
+                'Should never occurs, creator relationship led to null',
+            );
+        }
+
+        const createdInvitation = await MpeRoomInvitation.firstOrCreate({
+            mpeRoomID: roomID,
+            invitedUserID,
+            invitingUserID,
+        });
+
+        await room.related('invitations').save(createdInvitation);
+
+        const roomSummary: MpeRoomSummary = {
+            creatorName: room.creator.nickname,
+            isOpen: room.isOpen,
+            roomID: room.uuid,
+            roomName: room.name,
+            isInvited: true,
+        };
+
+        await UserService.emitEventInEveryDeviceUser(
+            invitedUserID,
+            'MPE_RECEIVED_ROOM_INVITATION',
+            [
+                {
+                    roomSummary,
+                },
+            ],
+        );
+    }
```

Comment:
If I understand correctly this function, we allow the creator to invite the same user several times?
I see that we were doing the same for MTV rooms.

Reply:
Yes, it has been confirmed by the test _It should create only one invitation in base and send MPE_RECEIVED_ROOM_INVITATION to the invited user clients_ you wrote 👌

Reply:
Yes has we don't retrieve in the client the already invited users for a given room we still want to allow a creator to send several toast to the invited user, in reality this could bring up a toast spam issue from a malicious user creator that would scan the db
This could be changed easily by avoiding sending a toast is the invitation is already existing
But should we ?

Reply:
At the moment I think it's perfect as is, but I was also thinking about this spamming issue.
Let's go on for now!